Realm/Realm/Handles/DictionaryHandle.cs

```diff
@@ -184,24 +184,17 @@ public bool TryGet(string key, RealmObjectBase.Metadata metadata, Realm realm, o
             }
```

Comment:
Can't comment on the right line, but you can probably combine both branches of the if statement since ToRealmValue will handle the `Type != Object` case anyway.